gapir/cc/interpreter.cpp

```diff
@@ -236,27 +337,82 @@ Interpreter::Result Interpreter::pushI(uint32_t opcode) {
     GAPID_WARNING("Error: pushI basic type invalid %d", (int)type);
     return ERROR;
   }
-  Stack::BaseValue data = extract20bitData(opcode);
+
+  uint64_t data = extract20bitData(opcode);
+
   switch (type) {
-    // Sign extension for signed types
+    case BaseType::Bool:
+      mStack.push(data != 0);
+      break;
+
+    case BaseType::Int8:
+      mStack.push<int8_t>(data);
+      break;
+
+    case BaseType::Int16:
+      mStack.push<int16_t>(data);
+      break;
+
     case BaseType::Int32:
+      data |= (data & 0x80000) != 0
+                  ? 0xfff00000ULL
+                  : 0x0;  // Sign extend the 20 bit immediate value
+      mStack.push(*((int32_t*)&data));
+      break;
+
     case BaseType::Int64:
-      if (data & 0x80000) {
-        data |= 0xfffffffffff00000ULL;
-      }
+      data |= (data & 0x80000) != 0
+                  ? 0xfffffffffff00000ULL
+                  : 0x0;  // Sign extend the 20 bit immediate value
+      mStack.push(*((int64_t*)&data));
+      break;
+
+    case BaseType::Uint8:
+      mStack.push<uint8_t>(data);
       break;
-    // Shifting the value into the exponent for floating point types
+
+    case BaseType::Uint16:
+      mStack.push<uint16_t>(data);
+      break;
+
+    case BaseType::Uint32:
+      mStack.push<uint32_t>(data);
+      break;
+
+    case BaseType::Uint64:
+      mStack.push<uint64_t>(data);
+      break;
+
     case BaseType::Float:
-      data <<= 23;
+      data <<=
```

Comment:
The alignment here is very odd.

Reply:
clang-format

Reply:
Can you move the comment to another line, so that it doesn't have this alignment issue